### PR TITLE
Update Gather-7 specification

### PR DESCRIPTION
**Pull request by achetver**
### Details:
 - Update Gather-7 operation specification since we decide to support negative values of `batch_dims` attribute. Also update formula to same view as in [TF](https://www.tensorflow.org/api_docs/python/tf/gather) specification to make it more clear.

### Tickets:
 - 54329

### CODE REVIEW

**Comment by achetver** [comment]
@sadolini @yekruglov @iimironov @lazarevevgeny Please, review this PR with important changes in Gather-7 operation specification.

**Review by yekruglov — APPROVED** [review]

**Comment by a-sidorova** [comment]
I think that description for `axis` is incorrect:
`Allowed values are from [-len(data.shape), len(indices.shape) - 1] and axis >= batch_dims`

My suggestion is:
`Allowed values are from [-len(data.shape), len(data.shape) - 1] and axis >= batch_dims`

Because otherwise the example 4 contradicts this description. `axis = 2` but `len(indices.shape) - 1 = 2 - 1 =1`.

**Comment by andrei-cv** [comment]
Allowed values are from [-len(indices.shape), len(indices.shape) - 1] and axis >= batch_dims as in tensorflow

File: docs/ops/movement/Gather_7.md [hunk 1]
```diff
@@ -10,18 +10,19 @@ TensorFlow\* [Gather](https://www.tensorflow.org/api_docs/python/tf/gather) oper
...
 
 **Attributes**:
 * *batch_dims*
   * **Description**: *batch_dims* (also denoted as `b`) is a leading number of dimensions of `data` tensor and `indices` 
   representing the batches, and *Gather* starts to gather from the `b` dimension. It requires the first `b` 
-  dimensions in `data` and `indices` tensors to be equal.
-  * **Range of values**: `[0; min(data.rank, indices.rank))` and `batch_dims <= axis`
+  dimensions in `data` and `indices` tensors to be equal. If *batch_dims* is less than zero used normalized value 
+  `batch_dims = indices.rank + batch_dims`.
+  * **Range of values**: `[-min(data.rank, indices.rank); min(data.rank, indices.rank))` and `batch_dims <= axis`
```
Comment (sadolini) [inline docs/ops/movement/Gather_7.md:25]: batch_dims<=axis is true for any batch_dims<0 and axis>0. Also should we require batch_dims<0 for axis<0?
 I think this condition should be apllied to normalized batch_dims value and abs(axis).

File: docs/ops/movement/Gather_7.md [hunk 2]
```diff
@@ -121,7 +140,7 @@ output_shape = (2, 1, 3, 4)
 
 * **3**:  Scalar or 1D tensor `axis` of *T_AXIS* type is a dimension index to gather data from. For example, 
 *axis* equal to 1 means that gathering is performed over the first dimension. Negative value means reverse indexing. 
-Allowed values are from `[-len(data.shape), len(indices.shape) - 1]` and `axis >= batch_dims`. 
+Allowed values are from `[-len(data.shape), len(data.shape) - 1]` and `axis >= batch_dims`.
```
Comment (sadolini) [inline docs/ops/movement/Gather_7.md:143]: The same comment for condition axis>=batch_dims as in batch_dims description.

File: docs/ops/movement/Gather_7.md (hunk 1 above)
Comment (pavel-esir) [reply docs/ops/movement/Gather_7.md:25]: you are right `batch_dims<=axis` is true either if both batch_dims and axis are of the same sight or if they are normalized. Will correct this

Comment (pavel-esir) [reply docs/ops/movement/Gather_7.md]: added clarification about this, please review

File: docs/ops/movement/Gather_7.md (hunk 2 above)
Comment (pavel-esir) [reply docs/ops/movement/Gather_7.md]: replied above

File: docs/ops/movement/Gather_7.md (hunk 1 above)
Comment (pavel-esir) [reply docs/ops/movement/Gather_7.md]: > Allowed values are from [-len(indices.shape), len(indices.shape) - 1] and axis >= batch_dims as in tensorflow

indeed there must be `data.shape` not indices. Axis value is not bound to indices. In TF for axis allowed values are 
Allowed values are from `[-len(data.shape), len(data.shape) - 1] and axis >= batch_dims`

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/gather_op.cc#L81-L89
there params goes for 1st input in our case means data

**Comment by pavel-esir** [comment]
> Allowed values are from [-len(indices.shape), len(indices.shape) - 1] and axis >= batch_dims as in tensorflow

indeed there must be `data.shape` not indices. Axis value is not bound to indices. In TF for axis allowed values are 
Allowed values are from `[-len(data.shape), len(data.shape) - 1] and axis >= batch_dims`

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/gather_op.cc#L81-L89
there params goes for 1st input in our case means data

**Review by sadolini — APPROVED** [review]

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**